packages/@aws-cdk/aws-events/lib/util.ts

```diff
@@ -38,9 +38,11 @@ export function mergeEventPattern(dest: any, src: any) {
           `Type mismatch between existing pattern ${JSON.stringify(destValue)} and added pattern ${JSON.stringify(srcValue)}`);
       }
 
-      // if this is an array, concat the values
+      // if this is an array, concat and deduplicate the values
       if (Array.isArray(srcValue)) {
-        destObj[field] = destValue.concat(srcValue);
+        const result = [...destValue, ...srcValue];
+        const resultJson = result.map(i => JSON.stringify(i));
+        destObj[field] = result.filter((value, index) => resultJson.indexOf(JSON.stringify(value)) === index);
```

Comment:
We use `indexOf` to find the `current value` from start. If the output does not equal to value `index`, it means there is a same value before current value and current value meed to remove.